### PR TITLE
week 4 authentication

### PR DESCRIPTION
**Pull request by ashleygranitto**
Here is my authentication feature! Sorry I did not realize how many files I added and modified until I started to commit all of them. Regarding my question during pod sync today, right now I was hoping to minimalist-ly style the text in UpdateLoginServlet.java, but trying to incorporate a CSS class into the HTML did not work.

### CODE REVIEW

**Comment by ashleygranitto** [comment]
I modified my login such that the servlet sends the relevant HTML back to an async JS function to be put into my new account.html page--now the user only must leave the page to go to the Google login page--there's no longer an intermediary step.

File: portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    UserService userService = UserServiceFactory.getUserService();
+    response.setContentType("application/json");
+
+    // Return a keyword based upon the user's login status
+    if (userService.isUserLoggedIn()) {
+      String json = convertToJson("in");
```
Comment (mmallett) [inline portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:37]: `"in"` isn't a proper JSON string, You don't need to convert it to JSON, I don't think this function would do anything.

Alternatively, you could return a proper JSON string

```
{ "isLoggedIn": true}
```

File: portfolio/src/main/webapp/comment-form.js [hunk 2]
```diff
@@ -1,5 +1,25 @@
...
+/** Update the page upon page load */
+function init() {
+  checkLogin();
+  getComments(); 
+}
+
+/** Ensure the comment form is only displayed if the user is logged in */
+async function checkLogin() { 
+  const response = await fetch('/login-status');
+  const status = await response.json();
```
Comment (mmallett) [inline portfolio/src/main/webapp/comment-form.js:12]: This is not actually JSON, so as it is now, it should be `response.text()`

**Review by mmallett — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java (hunk 1 above)
Comment (ashleygranitto) [reply portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java:37]: changed to proper JSON

File: portfolio/src/main/webapp/comment-form.js (hunk 2 above)
Comment (ashleygranitto) [reply portfolio/src/main/webapp/comment-form.js:12]: changed to proper JSON

**Pull request merged**